fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/OptOlapPartitionPruner.java

```diff
@@ -105,6 +105,37 @@ public static LogicalOlapScanOperator prunePartitions(LogicalOlapScanOperator lo
         return builder.build();
     }
 
+    /**
+     * If the selected partition ids are not null, merge the selected partition ids and do further partition pruning.
+     * @param olapScanOperator the olap scan operator
+     * @return the new olap scan operator with merged partition ids and predicates
+     */
+    public static LogicalOlapScanOperator mergePartitionPrune(LogicalOlapScanOperator olapScanOperator) {
+        // already pruned partitions
+        List<Long> selectedPartitionIds = olapScanOperator.getSelectedPartitionId();
+        // new pruned partitions
+        LogicalOlapScanOperator newOlapScanOperator = OptOlapPartitionPruner.prunePartitions(olapScanOperator);
+        // merge selected partition ids
+        List<Long> ansPartitionIds = null;
```

Comment:
The final result of twice partition pruning is intersection of both result; suggest to code as follows:
```java
Set<Long> result1 = Optional.ofNullable(olapScanOperator.getSelectedPartitionId())
               .map(ids->new HashSet(ids))
               .orElse(Collections.emptySet());
Set<Long> result2 = Optional.ofNullable(olapScanOperator.getSelectedPartitionId())
               .map(ids->new HashSet(ids))
               .orElse(Collections.emptySet());

List<Long> finalResult = Lists.newArrayList(Sets.intersect(result1,result2));
```

```